src/java.base/share/classes/java/lang/reflect/Proxy.java

```diff
@@ -1245,13 +1246,13 @@ public static InvocationHandler getInvocationHandler(Object proxy)
      *         <li>the given {@code method} is overridden directly or indirectly by
      *             the proxy interfaces and the method reference to the named
      *             method never resolves to the given {@code method}; or</li>
-     *         <li>if length of given {@code args} array doesn't match the number of
-     *             parameters of the method to be invoked or if {@code args} is null
-     *             and the method to be invoked has parameters</li>
-     *         <li>if any of the {@code args} elements can't be assigned to the
-     *             boxed type of the corresponding method parameter or any of the
-     *             {@code args} elements is null while the corresponding method
-     *             parameter is of primitive type</li>
+     *         <li>the length of the given {@code args} array does not match the
+     *             number of parameters of the method to be invoked; or</li>
+     *         <li>any of the {@code args} elements fails the unboxing
+     *             conversion if the corresponding method parameter type is
+     *             a primitive type; or if, after possible unboxing, any of the
+     *             {@code args} elements cannot be assigned to the corresponding
+     *             method parameter type.</li>
      *         </ul>
```

Comment:
Hm, well, on a second thought, the following: "...or if, after possible unboxing, any of the args elements cannot be assigned to the corresponding method parameter type"
For example, if the method parameter type is primitive long and an Integer object is passed to the invokeDefaultMethod at its place, then according to above wording "after unboxing" we get an int value. Can int value be assigned to long parameter? What happens is that Integer object 1st tries to be cast to Long and only then unboxing happens. Such is the order of spreadArguments MH operator.